src/noit_metric_tag_search.c

```diff
@@ -49,6 +49,41 @@ typedef struct noit_var_match_t {
 
 static mtev_boolean noit_match_str(const char *subj, int subj_len, struct noit_var_match_t *m);
 
+static pthread_mutex_t noit_tags_search_tls_jit_init_lock = PTHREAD_MUTEX_INITIALIZER;
+static __thread pcre_jit_stack *noit_tag_search_tls_jit_stack;
+static bool noit_tags_search_tls_jit_init = false;
+static pthread_key_t noit_tag_search_tls_jit_stack_key;
+
+static void
+noit_tag_search_free_tls_pcre_jit_stack(void *stack)
+{
+  pcre_jit_stack_free((pcre_jit_stack *)stack);
+}
+static pcre_jit_stack *
+noit_tag_search_get_tls_pcre_jit_stack(void *arg)
+{
+  (void)arg;
+  if (noit_tags_search_tls_jit_init && noit_tag_search_tls_jit_stack == NULL) {
```

Comment:
One of these is thread local and the other is not?  Doesn't the initialization need to happen once per thread?

Reply:
no...  the initialization just needs to happen once.  This skips attempts to do work if it isn't initialized.

There is a key and a __thread to avoid the cost of the _getspecific call once it is allocated on a thread.